### PR TITLE
Pseudo-invariant feature matching tutorial

### LINKED ISSUE #611: [Tutorial proposal] Pseudo-invariant feature matching in Earth Engine
**What is the objective of the proposed tutorial?**
Apply relative radiometric normalization using [pseudo-invariant feature matching](https://www.sciencedirect.com/science/article/abs/pii/0034425788901162).

**What is the scope of the proposed tutorial?**
A pair of public SkySat images acquired before and after forest conversion in Peru. JS API.

**Please provide an outline of the structure of the proposed tutorial?**
1. Intro
    1. Change detection and radiometric normalization
        1. Explanation and applications
        2. PIF compared to  [histogram matching](https://developers.google.com/earth-engine/tutorials/community/histogram-matching)  
    2. Loading data
        1. Before and after images from SkySat
3. Detecting Pseudo-Invariant Features
    1. Calculating spectral distance
    2. Identifying stable pixels
        1. Percentile reducer
4. Applying Normalization
    1. Band-wise linear fitting
        1. Mapping over bands and combining into an image 
6. Results

**In what format will you be submitting the tutorial?**
Markdown

Here's a draft of the [JS script](https://code.earthengine.google.com/b002f7ceec1c474682ee86811bd3800c) that I would be going over.

### PR DESCRIPTION
**Pull request by aazuspan**
Closes #611.

Here's the tutorial ready for review. Lint and tests are passing. 

Happy to make any changes!

### CODE REVIEW

File: tutorials/pseudo-invariant-feature-matching/index.md [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+
+### Selecting Stable Pixels
+
+The next step is to select a subset of pixels that experienced little or no
+change. This can be done by applying a percentile reducer over the area of
+interest to identify a spectral distance threshold, then selecting pixels with
+distances below that threshold. The code below uses a 10th percentile
+threshold, but you may want to modify the percentile to see how it affects
+results. Higher percentiles will select more pixels and include more
+variability, leading to more aggressive transformations.
```
Comment (jdbcode) [inline tutorials/pseudo-invariant-feature-matching/index.md:114]: Not sure that "more aggressive" is quite right. Maybe something like:

Higher percentiles will select more pixels with increasingly greater spectral distance, leading to transformations that are less weighted by PIFs.

^ That's probably too wordy ¯\_(ツ)_/¯  :)

**Review by jdbcode — COMMENTED** [review]
This is really great, Aaron! Thanks for the clean formatting and succinct text.

Just two minor comments.

Also, could you upload a `thumbnail.png` image that is 600x340 (width, height) for a tutorial card on [this page](https://developers.google.com/earth-engine/tutorials/community/explore).

Note that I made a change for formatting to the PR index.md file, so you may need to sync.

File: tutorials/pseudo-invariant-feature-matching/index.md [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+  .clip(aoi)
+  .register(before, 100);
+```
+
+Adding the images to the map shows the radiometric mismatch between them:
+the 'after' image appears much brighter than the 'before' image, making direct
+comparisons more difficult.
+
+```javascript
+Map.addLayer(before, {min: 1000, max: 5000}, 'Before');
```
[the remark this answers is not in the document]
Comment (aazuspan) [reply tutorials/pseudo-invariant-feature-matching/index.md:71]: Good point, fixed!

File: tutorials/pseudo-invariant-feature-matching/index.md (hunk 1 above)
Comment (aazuspan) [reply tutorials/pseudo-invariant-feature-matching/index.md]: Agreed! I somehow made it even more wordy, let me know what you think...

Comment (jdbcode) [reply tutorials/pseudo-invariant-feature-matching/index.md]: I like it!

**Comment by jdbcode** [comment]
Aaron, some things happen on our end now. It'll probably be published by the end of the week. I'll let you know here when it's live.

**Comment by aazuspan** [comment]
Thanks Justin!

**Comment by jdbcode** [comment]
The tutorial is live!
https://developers.google.com/earth-engine/tutorials/community/pseudo-invariant-feature-matching

Thanks for supporting the EE developer community!

If you want to make any changes in the future, just submit a pull request.
